File: LivrusVirtus/FoxToolKit/FXPython/Splash.py
from FXPy.fox import *
from RotinasGlobais import RotinasGlobais
import Login

def create(parent):
  return frmSplash(parent)

class frmSplash(FXDialogBox):
  ID_FRMSPLASH_TIMER = 100
  mTimer = None

  def __init__(self, owner):
    FXDialogBox.__init__(self,owner,"Splash", DECOR_NONE,
      0, 0, 0, 0, 0, 0, 0, 0, 4, 4)

    FXMAPFUNC(self,SEL_TIMEOUT, frmSplash.ID_FRMSPLASH_TIMER, frmSplash.onTimer)

    Rotinas = RotinasGlobais.Rotinas
    bmpLivrusVirtus = Rotinas.lerFigura(self.getApp(), "livrusvirtus.bmp")
    FXLabel(self, "", bmpLivrusVirtus, LAYOUT_CENTER_X)

  def create(self):
    FXDialogBox.create(self) 

  def iniciaTimer(self):
    self.mTimer = self.getApp().addTimeout(1000, self, frmSplash.ID_FRMSPLASH_TIMER)
  
  def onTimer(self, sender, sel, ptr):
    self.mTimer = None
    Login.fLogin.show(PLACEMENT_SCREEN)
    self.destroy()
    return True

File: LivrusVirtus/FoxToolKit/FXRuby/Splash.rb
require 'fox'
include Fox
require 'Login'

$frmSplash = nil

class FRMSplash < FXDialogBox

  ID_FRMSPLASH_TIMER = 100

  def initialize(owner)
    super(owner, "Splash", DECOR_NONE,
      0, 0, 0, 0, 0, 0, 0, 0, 4, 4)

    FXMAPFUNC(SEL_TIMEOUT, ID_FRMSPLASH_TIMER, :onTimer)

    bmpLivrusVirtus = $Rotinas.lerFigura(getApp(), "livrusvirtus.bmp")
    FXLabel.new(self, "", bmpLivrusVirtus, LAYOUT_CENTER_X)
  end
  
  def create
    super
  end 

  def iniciaTimer()
    @mTimer = getApp().addTimeout(1000, self, ID_FRMSPLASH_TIMER)
  end
  
  def onTimer(sender, sel, ptr)
    @mTimer = nil
    $frmLogin.show(PLACEMENT_SCREEN)
    self.destroy()
    return true
  end

end
